cli/preprocessor.go

```diff
@@ -119,35 +121,51 @@ func runCompilerExecutable(
 	modeSingle bool,
 	fileInput string,
 	dataInput string,
-	lineCommentChars string,
-	blockCommentCharsOpen string,
-	blockCommentCharsClose string,
-) (result string) {
+	lineCommentIden string,
+	blockCommentIdenOpen string,
+	blockCommentIdenClose string,
+) string {
+	// Determine executeable name by name of compiler
+	executableName := "./ccomc"
 	switch compiler {
 	case "cpp", "c++":
-		result = util.ExecuteAndWaitWithOutput("./ccomc", strconv.FormatBool(modeSingle), fileInput, dataInput, lineCommentChars, blockCommentCharsOpen, blockCommentCharsClose)
+		executableName = "./ccomc"
 	case "java":
-		result = util.ExecuteAndWaitWithOutput("./ccomc-java.jar", strconv.FormatBool(modeSingle), fileInput, dataInput, lineCommentChars, blockCommentCharsOpen, blockCommentCharsClose)
+		executableName = "./ccomc-java.jar"
 	default:
 		log.Fatal("Invalid compiler name. Only 'cpp' and 'java' are allowed values.")
 	}
-	return
+	// Check if executable exists
+	if !util.CommandExists(executableName) {
+		log.Fatal("Compiler executable not found. Please check your installation")
+	}
+	// Execute compiler with user inputs
+	return util.ExecuteAndWaitWithOutput(executableName, strconv.FormatBool(modeSingle), fileInput, dataInput, lineCommentIden, blockCommentIdenOpen, blockCommentIdenClose)
 }
 
-func getCommentCharsFromLang(lang string) (lineCommentChars string, blockCommentCharsOpen string, blockCommentCharsClose string) {
+func getCommentIdenFromLang(lang string, fileInput string) (lineCommentIden string, blockCommentIdenOpen string, blockCommentIdenClose string) {
+	if lang == "auto" {
+		if !util.FileExists(fileInput) {
+			log.Fatal("Please use lang 'auto' only in combination of valid file paths as file input")
```

Comment:
Does `log.Fatal(...)` terminate or is it just writing to the log?

Reply:
`log.Fatal` termintates.